iqm5q.py

```diff
@@ -17,65 +20,79 @@
 
 TWPA_ADDRESS = "192.168.0.35"
 
+N_QUBITS = 5
+
 
 def create(runcard_path=RUNCARD):
     """IQM 5q-chip controlled Zurich Instrumetns (Zh) SHFQC, HDAWGs and PQSC.
 
     Args:
         runcard_path (str): Path to the runcard file.
     """
-    # Instantiate Zh set of instruments[They work as one]
-    instruments = {
-        "SHFQC": [{"address": "DEV12146", "uid": "device_shfqc"}],
-        "HDAWG": [
-            {"address": "DEV8660", "uid": "device_hdawg"},
-            {"address": "DEV8673", "uid": "device_hdawg2"},
-        ],
-        "PQSC": [{"address": "DEV10055", "uid": "device_pqsc"}],
-    }
 
-    shfqc = []
-    for i in range(5):
-        shfqc.append(
-            {"iq_signal": f"q{i}/drive_line", "ports": f"SGCHANNELS/{i}/OUTPUT"}
-        )
-        shfqc.append(
-            {"iq_signal": f"q{i}/measure_line", "ports": ["QACHANNELS/0/OUTPUT"]}
-        )
-        shfqc.append(
-            {"acquire_signal": f"q{i}/acquire_line", "ports": ["QACHANNELS/0/INPUT"]}
-        )
-
-    hdawg = []
-    for i in range(5):
-        hdawg.append({"rf_signal": f"q{i}/flux_line", "ports": f"SIGOUTS/{i}"})
-    for c, i in zip(itertools.chain(range(0, 2), range(3, 4)), range(5, 8)):
-        hdawg.append({"rf_signal": f"qc{c}/flux_line", "ports": f"SIGOUTS/{i}"})
-
-    hdawg2 = [{"rf_signal": "qc4/flux_line", "ports": f"SIGOUTS/0"}]
-
-    pqsc = [
-        "internal_clock_signal",
-        {"to": "device_hdawg2", "port": "ZSYNCS/1"},
-        {"to": "device_hdawg", "port": "ZSYNCS/0"},
-        {"to": "device_shfqc", "port": "ZSYNCS/2"},
-    ]
+    device_setup = DeviceSetup("EL_ZURO")
+    # Dataserver
+    device_setup.add_dataserver(host="localhost", port=8004)
+    # Instruments
+    device_setup.add_instruments(
+        HDAWG("device_hdawg", address="DEV8660"),
+        HDAWG("device_hdawg2", address="DEV8673"),
+        PQSC("device_pqsc", address="10055"),
+        SHFQC(
+            "device_shfqc", address="device_shfqc", reference_clock_source="internal"
```

Comment:
```suggestion
            "device_shfqc", address="DEV12146", reference_clock_source="internal"
```

Reply:
I mistakenly included `reference_clock_source` for shfqc, so fixing the device address in a separate suggestion.